### DISCUSSION TITLE
OAuth Provider: "response" is not a conform Token Endpoint response

### DISCUSSION THREAD

**Discussion by marmundo** [nextauthjs/next-auth#10646] [Help]
I'm building a provider to my university oauth. However when app gets code from university using callback url, I get 

```
[auth][error] CallbackRouteError: Read more at https://errors.authjs.dev#callbackrouteerror
[auth][cause]: OperationProcessingError: "response" is not a conform Token Endpoint response
```
I saw this problem caused by oauth4webapi, however I can't track what the problems with request to acess_token.

My questions are:
1- Is there a way to see what headers, payload inside next auth. Like see more detailed log.
2- What is a conform token endpoint response to check if my university token endpoint is returning in a proper format?

Thanks.

**Comment by timfee** [comment]
I am experiencing the same with OneLogin

**Comment by timfee** [comment]
Ah, if it helps, my base URL ("issuer") was `https://xxx.onelogin.com` in v4. 
In v5 i had to specify `https://xxx.onelogin.com/oidc/2` manually

**Comment by marmundo** [comment]
Hi,

Does this change makes it work?

**Reply by leonwangcn** [reply]
You can refer to
https://github.com/nextauthjs/next-auth/pull/13259/commits/85d55a7ecc8d7ac5626079047d33c1cc30e6b46d

**Comment by leonwangcn** [comment]
"scripts": {
"postinstall": "prisma generate && node scripts/patch-auth-core.cjs",
...
}
[scripts.zip](https://github.com/user-attachments/files/22592488/scripts.zip)
